File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/comment/comment/direct/Java/Python/s378126458_commented.py
import sys

def main():
    input = sys.stdin.read
    data = input().split()
    
    idx = 0
    N = int(data[idx])
    Q = int(data[idx+1])
    idx += 2
    
    s = data[idx]
    idx += 1
    
    pre = [0] * (N + 1)
    for i in range(1, N):
        if s[i-1] == 'A' and s[i] == 'C':
            pre[i] = 1  # initially the prefix sum array is the k array
    
    # Building the psa
    for i in range(1, N):
        pre[i] += pre[i-1]
    
    # Querying
    for _ in range(Q):
        l = int(data[idx]) - 1
        r = int(data[idx+1]) - 1
        idx += 2
        print(pre[r] - pre[l])

if __name__ == "__main__":
    main()  # End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/multiline/comment/direct/Java/C/s378126458_commented.c
#include <stdio.h>
#include <string.h>

int main() {
    int N, Q;
    scanf("%d %d", &N, &Q);
    char s[100005];
    scanf("%s", s);
    int pre[100005] = {0};
    
    for(int i = 1; i < N; i++) {
        if(s[i-1] == 'A' && s[i] == 'C') {
            pre[i] = 1; // initially the prefix sum array is the k array
        }
    }
    
    // Building the psa
    for(int i = 1; i < N; i++) {
        pre[i] += pre[i-1];
    }
    
    // Querying
    while(Q--) {
        int l, r;
        scanf("%d %d", &l, &r);
        r--;
        printf("%d\n", pre[r] - pre[l-1]);
    }
} // End of Code
